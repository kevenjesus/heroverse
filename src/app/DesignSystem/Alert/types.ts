import { ReactNode } from "react"

export interface WrapperTypes {
    type: 'error' | 'success'
}

export interface AlertProps extends WrapperTypes {
    children: ReactNode
}