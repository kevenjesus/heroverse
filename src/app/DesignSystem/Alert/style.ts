import styled from 'styled-components'
import { WrapperTypes } from './types'
import { colors } from '@/app/styles/guide'

export const Wrapper = styled.div<WrapperTypes>`
    width: 100%;
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 16px;
    padding: 0 20px;
    background: ${({type}) => type === 'error' ? colors.alertError : colors.alertSuccess};
`