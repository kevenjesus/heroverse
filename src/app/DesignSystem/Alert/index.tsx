import Typographic from '../Typographic'
import * as S from './style'
import { AlertProps } from './types'

export default function Alert({children, type}: AlertProps) {
    return (
        <S.Wrapper type={type}>
             <img width={28} src={type === 'error' ? '/warning.png' : '/checked.png'} alt={`${children}`} />
            <Typographic color='white' weight='700'>{children}</Typographic>
        </S.Wrapper>
    )
}