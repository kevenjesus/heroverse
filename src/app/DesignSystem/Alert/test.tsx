import React from 'react';
import { render } from '@testing-library/react';
import Alert from '.';
import '@testing-library/jest-dom'


describe('Alert component', () => {
    it('should render with the type Success', () => {
        const { container } = render(<Alert type="success">Test alert</Alert>);
        expect(container).toMatchSnapshot()
    });

    it('should render with the type Success', () => {
        const { getByRole } = render(<Alert type="success">Test alert</Alert>);
        const img = getByRole('img');
        expect(img).toHaveAttribute('src', '/checked.png');
    });

    it('should render with the type Error', () => {
        const { getByRole } = render(<Alert type="error">Test alert</Alert>);
        const img = getByRole('img');
        expect(img).toHaveAttribute('src', '/warning.png');
    });

});
