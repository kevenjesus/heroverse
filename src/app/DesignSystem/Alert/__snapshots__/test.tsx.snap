// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render with the type Success 1`] = `
<div>
  <div
    class="sc-hLseeU fgkOnj"
    type="success"
  >
    <img
      alt="Test alert"
      src="/checked.png"
      width="28"
    />
    <p
      class="sc-dmqHEX iEKdJV"
      color="white"
    >
      Test alert
    </p>
  </div>
</div>
`;
